### PR TITLE
Adds an autoclave flag to item::make.

### PR DESCRIPTION
**Pull request by wito**
This allows things like sealed jar of pickled food to better reset their container when transforming, avoiding things like 'glass jar (1)'.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Does this also address musical instruments i.e. ````bagpipes (1)````?

**Comment by wito** [comment]
> On 20 Oct 2015, at 20:23, Angela Graves <notifications@github.com> wrote:
> 
> Does this also address musical instruments i.e. bagpipes (1)?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/CleverRaven/Cataclysm-DDA/pull/13790#issuecomment-149655054>.
> 

Instruments are afflicted by a separate issue.


Submitting separate PR to fix.

**Comment by BevapDin** [comment]
An alternative solution to the "(1)" on sealed jars:

Change the item type of the sealed jar to "GENERIC" (remove the tool-specific members as well), like it is for the "sealed jar of pickled meat" (tools.json). It seems only the "fermenting sauerkraut jar" has currently this "(1)" problem, the other sealed jars work fine for me (without the "(1)" prefix).

**Comment by wito** [comment]
> On 20 Oct 2015, at 20:33, BevapDin <notifications@github.com> wrote:
> 
> An alternative solution to the "(1)" on sealed jars:
> 
> Change the item type of the sealed jar to "GENERIC" (remove the tool-specific members as well), like it is for the "sealed jar of pickled meat" (tools.json). It seems only the "fermenting sauerkraut jar" has currently this "(1)" problem, the other sealed jars work fine for me (without the "(1)" prefix).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/CleverRaven/Cataclysm-DDA/pull/13790#issuecomment-149658502>.
> 
In my testing this affects all kinds of fermenting items with containers; sauerkraut, yeast, etc.; (glass flasks in the case of yeast). The jars aren’t created through crafting, but through delayed_transform, so I’m not sure the GENERIC tag would do any good.

**Comment by wito** [comment]
The second commit isn't _strictly_ speaking related, but since endless nested components being stored was part of the discovery, it was put in there.

**Comment by BevapDin** [comment]
> In my testing this affects all kinds of fermenting items with containers; sauerkraut, yeast, etc.; (glass flasks in the case of yeast).

Ah, yes, "flask_yeast" is also a tool. But have you *seen* this bug with the other sealed jars (meat, pickled vegetable/meat, soups)? Because I don't see it on those.

> The jars aren’t created through crafting, but through delayed_transform, so I’m not sure the GENERIC tag would do any good.

Technically the item itself is created through crafting, and that is where it gets the 1 charge (or no charge) from. Later is just changed to the jar and the charge remain, but you know that because you've seen the code. What I meant is: changing the item type to GENERIC (not TOOL) will make the initially created item have not charges. And consequently the jar (after the transformation) will not have charges either.

**Comment by wito** [comment]
That might work, yeah.

However, part of what I discovered was that charges or no, the jars still build up cruft (which is in part what the second commit is intended to resolve), and that still needs to be stripped. Will do some testing with GENERIC, tho'...

File: data/json/items/tools.json [hunk 1]
```diff
@@ -7438,7 +7438,7 @@
   
   {
     "id": "sauerkraut_ferment",
-    "type": "TOOL",
+    "type": "GENERIC",
```
Comment (BevapDin) [inline data/json/items/tools.json]: Please also remove the TOOL specific members "max_charges", "initial_charges", "charges_per_use", "turns_per_charge", "ammo" and "revert_to". They have no meaning in a GENERIC item type.

There are some items in the More_Survival_Tools and blazemod mod that have the same problem.

Comment (wito) [inline data/json/items/tools.json]: It is done.

**Pull request merged**